### PR TITLE
Task4 branch

### PR DESCRIPTION
**Pull request by SiarheiPatapenka**
Hi Dzmitry , could you please have a look at my PR? Thank you!

Test results are in the screenshots:
image
image

### CODE REVIEW

File: tasks/arrays/arrays.js [hunk 1]
```diff
@@ -1,57 +1,89 @@
...
 /**
  * print (console.log) ids of all characters
+ * @param {Array} chars
  */
 function printCharacterNames(chars) {
-}
-
+   
+    for ( let element of chars)  {
+      console.log(element.name );
```
Comment (dshaplyko) [inline tasks/arrays/arrays.js:25]: Nice. It can be also done by forEach() method

File: tasks/arrays/arrays.js [hunk 2]
```diff
@@ -1,57 +1,89 @@
...
+ * @param {Array} chars
+ * @return {Array} - char names
  */
-function getCharactersNames(chars) {
-}
+
+ function getCharactersNames(chars) {
+    let masname = [];
+    for ( let element of chars)  {
+     masname.push(element.name);
```
Comment (dshaplyko) [inline tasks/arrays/arrays.js:10]: Nice! It can also be done by map() method :)

File: tasks/arrays/arrays.js [hunk 3]
```diff
@@ -1,57 +1,89 @@
...
+    return chars.find(el => (el.name == 'Jerry Smith'));
 }
 
 /**
  * check if all characters are human. return true if all, false if not
+ * @param {Array} chars
+ * @return {boolean}
  */
 function isAllHuman(chars) {
+    return chars.includes('Human');
```
Comment (dshaplyko) [inline tasks/arrays/arrays.js:54]: This method only checks if Human exists in the whole JSON file, and obviously this is true :). But you need to check if ALL characters are human, so it should be done by every() method.

File: tasks/arrays/arrays.js [hunk 4]
```diff
@@ -1,57 +1,89 @@
...
+ * @return {boolean}
  */
 function isAnyFishPerson(chars) {
-}
+    let masFish = [];
+    for ( let element of chars)  {
+     masFish.push(element.type = "Fish-Person");   
+      }     
+    
+      return  masFish.includes('Fish-Person');
```
Comment (dshaplyko) [inline tasks/arrays/arrays.js:68]: The same as the above. some() method is more appropriate here.

File: tasks/strings/strings.js [hunk 5]
```diff
@@ -1,31 +1,100 @@
...
+ */
+function reverseString(str) {
+  if (str >= 0) {
+    return 'This is not a string!';
+}
+
+else {
+    let elements = str.split('');
+elements.reverse();
+    return elements.join('');
```
Comment (dshaplyko) [inline tasks/strings/strings.js:14]: it can be done in one line:
elements = str.split('').reverse().join('');

File: tasks/arrays/arrays.js (hunk 3 above)
Comment (SiarheiPatapenka) [reply tasks/arrays/arrays.js]: Done by using every() method.

File: tasks/arrays/arrays.js (hunk 4 above)
Comment (SiarheiPatapenka) [reply tasks/arrays/arrays.js]: Done by using some() method.

**Pull request merged**